src/rdkafka_broker.c

```diff
@@ -5913,6 +5912,31 @@ rd_kafka_broker_update (rd_kafka_t *rk, rd_kafka_secproto_t proto,
 }
 
 
+/**
+ * @returns the broker id, or RD_KAFKA_NODEID_UA if \p rkb is NULL.
+ *
+ * @locality any
+ * @locks_required none
+ * @locks_acquired rkb_lock
+ */
+int32_t rd_kafka_broker_id (rd_kafka_broker_t *rkb) {
+        int32_t broker_id;
+
+        if (unlikely(!rkb))
+                return RD_KAFKA_NODEID_UA;
+
+        /* Avoid locking if already on the broker thread */
```

Comment:
first instinct is locking may be better outside (simplicity over terseness). didn't think about it much.

Reply:
The entire point of this method is to abstract that detail away from the caller :)